clientgui/DlgAdvPreferencesBase.cpp

```diff
@@ -161,13 +161,16 @@ CDlgAdvPreferencesBase::CDlgAdvPreferencesBase( wxWindow* parent, int id, wxStri
 
     m_btnOK = new wxButton( m_panelButtons, wxID_OK, _("Save"), wxDefaultPosition, wxDefaultSize, 0 );
     m_btnOK->SetToolTip( _("Save all values and close the dialog.") );
-    
+    if (m_bUsingLocalPrefs) {
+        m_btnOK->SetDefault();
+    }
     buttonSizer->Add( m_btnOK, 0, wxALL, 5 );
 
     m_btnCancel = new wxButton( m_panelButtons, wxID_CANCEL, _("Cancel"), wxDefaultPosition, wxDefaultSize, 0 );
     m_btnCancel->SetToolTip( _("Close the dialog without saving.") );
-    m_btnCancel->SetDefault();
-
+    if (!m_bUsingLocalPrefs) {
+        m_btnCancel->SetDefault();
+    }
```

Comment:
Can this block be combined with the block on line 164 to be smth like this:

```
if (m_bUsingLocalPrefs) {
    m_btnOK->SetDefault();
} else {
    m_btnCancel->SetDefault();
}
```

I think this will make the code more clear for understanding

Reply:
I tried that first of all, but it didn't work. My conclusion was that the SetDefault action had to take place between the 'new' and 'add' actions. In the code we both inherited, all the actions for each of the three buttons were grouped together: OK/Save, Cancel, Help each went through their new/tooltip/default/add sequence. To use if...else  (it appeared to me) that we would have to re-sequence all the actions (do all the news, then all the sets, then all the adds) which was a bigger change than I felt confident making.

Reply:
Sorry, I didn't know about this. So I'm OK with this.